libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -1505,6 +1628,283 @@ void AP_CRSF_Telem::process_param_write_frame(ParameterSettingsWriteFrame* write
 #endif  // OSD_PARAM_ENABLED
 }
 
+#if AP_CRSF_SCRIPTING
+// scripting interface to parameter menus
+uint8_t AP_CRSF_Telem::get_menu_event(uint8_t menu_events, uint8_t& param_id, ScriptedPayload& payload)
+{
+    uint8_t events = 0;
+    ScriptedParameterWrite spw;
+    if (inbound_params.pop(spw)) {
+        // not listening for the event, send a response directly
+        if ((spw.type & menu_events) == 0) {
+            send_response(spw);
+            return 0;
+        }
+        switch (spw.type) {
+        case ScriptedParameterEvents::PARAMETER_WRITE:
+            events |= PARAMETER_WRITE;
+            payload = spw.payload;
+            param_id = spw.param->id;
+            break;
+        case ScriptedParameterEvents::PARAMETER_READ:
+            events |= PARAMETER_READ;
+            payload.payload_length = 0;
+            param_id = spw.param->id;
+            break;
+        default:
+            break;
+        }
+        // respond with parameter entry
+        outbound_params.push(spw);
+    }
+
+    return events;
+}
+
+bool AP_CRSF_Telem::send_write_response(uint8_t length, const char* data)
+{
+    ScriptedParameterWrite spw;
+    if (outbound_params.pop(spw)) {
+        switch (spw.type) {
+        case ScriptedParameterEvents::PARAMETER_READ:
+            // respond with parameter entry updated with the new data
+            spw.param->data = (const char*)hal.util->std_realloc((char*)spw.param->data, length);
+            memcpy((char*)spw.param->data, data, length);
```

Comment:
memcpy's second argument should generally by the length of the first argument, not the second argument!

Reply:
This shouldn’t need to be a realloc either. That function is on my chopping block, please free the old pointer then malloc a new one.

Reply:
... at an increase in flash cost, so I define a function to do it and we are back where we started. The future removal of something should not be a precondition for a PR - we can discuss separately. I *do* need to reallocate the memory to a difference size and realloc() is the standard way of doing that.

@peterbarker I don't get your comment - we have just reallocated the target to match the incoming length

Reply:
Fixed thanks